docs/xla_flags.md

```diff
@@ -0,0 +1,90 @@
+# XLA & Jax
+
+<!--* freshness: { reviewed: '2024-08-18' } *-->
+
+## Introduction
+This guide explains basics about XLA and how XLA related to Jax.
+For in-depth details please refer to [XLA documentation](https://openxla.org/xla)
+
+## XLA: The Powerhouse Behind JAX
+XLA (Accelerated Linear Algebra) is a domain-specific compiler for linear algebra that plays a pivotal role in JAX's performance and flexibility. It enables JAX to generate optimized code for various hardware backends (CPUs, GPUs, TPUs) by transforming and compiling your Python/NumPy-like code into efficient machine instructions.
+
+JAX uses XLA's JIT compilation capabilities to transform your Python functions into optimized XLA computations at runtime.
+
+## Configuring XLA in JAX:
+You can influence XLA's behavior in JAX by setting XLA_FLAGS environment variables before running your Python script or colab notebook.
+
+For the colab notebooks:
+
+Provide flags using `os.environ['XLA_FLAGS']`:
+
+
+```python
+import os
+
+# Set multiple flags separated by spaces
+os.environ['XLA_FLAGS'] = '--flag1=value1 --flag2=value2'
+```
+
+For the python scripts:
+
+Specify `XLA_FLAGS` as a part of cli command:
+
+```bash
+XLA_FLAGS='--flag1=value1 --flag2=value2'  python3 source.py
+```
+
+**Important Notes:**
+
+* Set XLA_FLAGS before importing JAX or other relevant libraries.
```

Comment:
I'd explicitly note that changing flags midway through a program (after backend initialization) will have no effect.